### ISSUE TITLE
cas_server.utils.update_url throws UnicodeEncodeError when updating a service URL with not purely ascii GET parameters

### ISSUE THREAD

**Issue by tobast** [nitmir/django-cas-server#82]
When requesting a login for a service URL that is not purely ASCII, say, `https://example.com/?next=/é` (properly URL-encoded in practice to `https%3A%2F%2Fexample.com%2F%3Fnext%3D%252F%25C3%25A9`), the `cas_server` fails with 500 error. This error is caused by `cas_server.utils.update_url`, calling `urllib.parse.parse_qsl` on the *utf-8 encoded GET arguments part of the URL*. This function fails with eg. `UnicodeEncodeError: 'ascii' codec can't encode character '\xe9' in position 1: ordinal not in range(128)`.

Digging into `urllib` code, this happens because `urllib` tries to coerce its return value to the same type as its input type, or, by default, `ascii`. The input value passed being bytes, it tries to coerce it to ascii by default, resulting in an error in the present case.

## Steps to reproduce

Start locally a django-cas-server, then query eg. `http://localhost:8000/login?service=https%3A%2F%2Fexample.com%2F%3Fnext%3D%252F%25C3%25A9`

## Possible fix

Unless this is required for some reason, avoiding conversion to bytes [here](https://github.com/nitmir/django-cas-server/blob/master/cas_server/utils.py#L253) seems to fix the issue. However, I assume this might be needed with older versions of urllib.

**Comment by nitmir** [comment]
Hi,

Thank for reporting.

I can reproduce. 
I suspect all this bytes testing is coming from previous python2 compatibility. I'll try to get read of all of it (and finally completely drop python2 support) and let Django deal with the unicode sandwich.

**Comment by nitmir** [comment]
Hi

This will be fix once a82d348b2078393eeb1fa6bab10a6c219228cd76 is merged into master.

Thanks for reporting.
